lua/barbar/state.lua

```diff
@@ -296,5 +324,41 @@ function state.icons(bufnr, activity)
   return buffer_icons
 end
 
+-- Save/load state
+
+--- Save recently_closed list
+--- @return nil
+function state.save_recently_closed()
+  local filepath = stdpath('cache') .. '/barbar.json'
+  local file = io.open(filepath, 'w+')
+  if file == nil then
+    return
+  end
+
+  local encoded_state = json_encode({
+    recently_closed = state.recently_closed,
+  })
+
+  file:write(encoded_state)
+  file:close()
+end
+
+--- Save recently_closed list
+--- @return nil
+function state.load_recently_closed()
+  local filepath = stdpath('cache') .. '/barbar.json'
+  local file = io.open(filepath, 'r')
+  if (file == nil) then
+    return
+  end
+  local content = file:read('*a')
+  file:close()
+
+  local saved_state = json_decode(content)
```

Comment:
```suggestion  
  local file, open_err = io.open(filepath, 'r')
  if open_err ~= nil then
    return utils.notify('Recently closed buffer list does not exist; skipping. ' .. filepath, vim.log.levels.INFO)
  elseif file == nil then
    return utils.notify('Could not open ' .. filepath, vim.log.levels.ERROR)
  end

  local content, read_err = file:read('*a')
  if read_err ~= nil then
    return utils.notify(read_err, vim.log.levels.ERROR)
  end

  local success, close_err = file:close()
  if close_err ~= nil then
    return utils.notify(open_err, vim.log.levels.ERROR)
  elseif success == false then
    return utils.notify('Could not close ' .. filepath, vim.log.levels.ERROR)
  end

  local saved_state = json_decode(content, {luanil = {array = true, object = true}})
```

Same as above, checking for errors. 

I made one of them an `INFO` because it _will_ happen the first time someone installs this plugin _or_ loads it up after downloading this update. If you just want to use `return` without a message that's fine too— but it could also be an indicator that something is wrong so I thought it might be worth putting a message there. Ultimately up to you.